mne/channels/montage.py

```diff
@@ -141,6 +141,12 @@ def read_montage(kind, ch_names=None, path=None, unit='m', transform=False):
         If not all electrodes defined in the montage are present in the EEG
         data, use this parameter to select subset of electrode positions to
         load. If None (default), all defined electrode positions are returned.
+
+        .. note:: ``ch_names`` are compared to channel names in the montage
+                  files after converting them both to upper case. If a match is
+                  found, the channel name in ``ch_names`` is included to the
+                  returned montage.
```

Comment:
Reading this again, I'm not sure if this is clear enough. Feel free to suggest something better.